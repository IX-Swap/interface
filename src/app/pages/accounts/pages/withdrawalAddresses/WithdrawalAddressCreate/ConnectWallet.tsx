import React, { ReactElement } from 'react'
import { Box, Button, Typography, CircularProgress } from '@mui/material'
import { useTheme } from '@mui/material/styles'
import { useWeb3React } from '@web3-react/core'
import { injected, coinbase } from 'config/blockchain/connectors'
import { SUPPORTED_WALLETS } from 'config/blockchain/supportedWallets'
import { useServices } from 'hooks/useServices'
import { useFormContext } from 'react-hook-form'
import { WithdrawalAddressFormValues } from 'types/withdrawalAddress'

interface ConnectWalletButtonProps {
  children: ReactElement
  isLoading?: boolean | undefined
}

const ConnectWalletButton = ({
  children,
  isLoading = false,
  ...rest
}: ConnectWalletButtonProps) => {
  const theme = useTheme()

  return (
    <Button
      fullWidth
      variant='contained'
      disableElevation
      disabled={isLoading}
      sx={{
        backgroundColor: theme.palette.paginationItem.borderHover,
        color: theme.palette.primary.main,
        paddingY: '18px'
      }}
      {...rest}
    >
      <Box display={'flex'} alignItems={'center'} gap={1.5}>
        {children}
        {isLoading && <CircularProgress color='inherit' size={'1rem'} />}
      </Box>
    </Button>
  )
}

export const ConnectWallet = ({
  onConnect,
  isLoading = false
}: {
  onConnect: Function
  isLoading?: boolean
}) => {
  const { activate } = useWeb3React()
  const { snackbarService } = useServices()

  const { control } = useFormContext<WithdrawalAddressFormValues>()

  const hasMetamaskExtension = window.ethereum?.isMetaMask === true

  // * Remove checking for Coinbase wallet extension (will default to pop-up with options to connect; including the QR option)
  //   const hasCoinbaseExtension = true

  // ! Doesn't work if user has no Metamask wallet extension enabled
  //   const hasCoinbaseExtension =
  //     typeof window.ethereum?.providers?.find(
  //       (provider: any) => provider.isCoinbaseWallet === true
  //     ) !== 'undefined'

  // * Seem to work even without the Metamask wallet extension enabled
  const hasCoinbaseExtension =
    window.web3?.currentProvider?.isCoinbaseWallet === true

  const failedToDetectExtension = () =>
    snackbarService.showSnackbar(
      <>
        Failed to detect wallet extension.
        <br />
        <br />
        Please install the correct wallet extension in your browser.
      </>,
      'error'
    )

  return (
    <Box
      display={'flex'}
      flexDirection={'column'}
      justifyContent={'center'}
      alignItems={'center'}
      width={'100%'}
      gap={2}
    >
      <ConnectWalletButton
        onClick={async () => {
          if (hasMetamaskExtension) {
            await activate(injected)
            control.setValue('wallet', 'METAMASK')
            onConnect()
          } else {
            failedToDetectExtension()
          }
        }}
        isLoading={isLoading}
      >
        <Box display={'flex'} alignItems={'center'} gap={1}>
          <span>MetaMask Wallet</span>
          <img
            src={SUPPORTED_WALLETS.METAMASK.iconURL}
            alt={'Metamask'}
            height={'20'}
          />
        </Box>
      </ConnectWalletButton>

      <Typography variant='subtitle2'>or</Typography>

      <ConnectWalletButton
        onClick={async () => {
          if (hasCoinbaseExtension) {
            await activate(coinbase)
            control.setValue('wallet', 'COINBASE')
            onConnect()
          } else {
            failedToDetectExtension()
          }
        }}
        isLoading={isLoading}
      >
        <Box display={'flex'} alignItems={'center'} gap={1}>
          <span>Coinbase Wallet</span>
          <img
            src={SUPPORTED_WALLETS.COINBASE.iconURL}
            alt={'Coinbase'}
            height={'20'}
          />
        </Box>
      </ConnectWalletButton>
    </Box>
  )
}

export const WalletInfo = ({ wallet }: { wallet: string }) => {
  if (wallet in SUPPORTED_WALLETS) {
    return (
      <Box
        display={'flex'}
        flexDirection={'column'}
        justifyContent={'center'}
        alignItems={'center'}
        padding={2}
        sx={{
          backgroundColor: '#F7F9FA',
          border: '1px solid #DBE2EC',
          borderRadius: '8px',
          gap: 1,
          marginBottom: '30px'
        }}
      >
        <Typography color={'#778194'} fontSize={14}>
          Connected to
        </Typography>
        <Box display={'flex'} gap={1} alignItems={'center'}>
          <Typography
            color={'##343A47'}
            fontSize={20}
            sx={{ textTransform: 'capitalize' }}
            variant='subtitle2'
          >
            {wallet?.toLowerCase()}
          </Typography>
          <img
            src={SUPPORTED_WALLETS[wallet].iconURL}
            alt={'Metamask'}
            height={'20'}
          />
        </Box>
      </Box>
    )
  }

  return <></>
}
