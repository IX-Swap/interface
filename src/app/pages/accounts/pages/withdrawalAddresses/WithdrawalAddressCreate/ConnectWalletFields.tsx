import { useFormContext } from 'react-hook-form'
import { WithdrawalAddressFormValues } from 'types/withdrawalAddress'
import { TypedField } from 'components/form/TypedField'
import { WalletSelect } from 'components/form/WalletSelect'
import { Grid } from '@material-ui/core'

export const ConnectWalletFields = () => {
  const { control } = useFormContext<WithdrawalAddressFormValues>()

  return (
    <>
      <Grid item>
        <TypedField
          component={WalletSelect}
          name='wallet'
          label='Select Wallet'
          control={control}
        />
      </Grid>
    </>
  )
}
