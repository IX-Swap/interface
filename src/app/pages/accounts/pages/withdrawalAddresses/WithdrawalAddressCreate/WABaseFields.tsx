import { Grid } from '@material-ui/core'
import { NetworkSelect } from 'components/form/NetworkSelect'
import { Radios } from 'components/form/Radios'
import { TypedField } from 'components/form/TypedField'
import { useFormContext } from 'react-hook-form'
import {
  BlockchainAddressVariant,
  WithdrawalAddressFormValues
} from 'types/withdrawalAddress'
import React from 'react'

const radios: Array<{ label: string; value: BlockchainAddressVariant }> = [
  {
    label: 'Connect to Wallet',
    value: 'connect'
  },
  {
    label: 'Input Blockchain Address',
    value: 'create'
  }
]

export const WABaseFields = () => {
  const { control, watch } = useFormContext<WithdrawalAddressFormValues>()
  const wallet = watch('wallet')
  const hasWallet = wallet !== undefined

  return (
    <>
      <Grid item>
        <TypedField
          customRenderer
          control={control}
          component={Radios}
          name='variant'
          label='Wallet Type'
          items={radios}
          style={{
            flexDirection: 'row',
            justifyContent: 'space-between'
          }}
        />
      </Grid>

      <Grid item>
        <TypedField
          control={control}
          component={NetworkSelect}
          name='network'
          label='Blockchain Network'
          disabled={hasWallet}
          variant='outlined'
        />
      </Grid>
    </>
  )
}
