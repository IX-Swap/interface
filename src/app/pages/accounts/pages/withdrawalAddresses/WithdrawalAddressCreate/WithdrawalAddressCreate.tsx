import React from 'react'
import { WADialog } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WADialog/WADialog'
import { WADialogTitle } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WADialog/WADialogTitle'
import { WADialogContent } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WADialog/WADialogContent'
import { CreateWalletDialog } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/CreateWalletDialog/CreateWalletDialog'
import { CreateWalletDialogTitle } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/CreateWalletDialog/CreateWalletDialogTitle'
import { CreateWalletDialogContent } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/CreateWalletDialog/CreateWalletDialogContent'
import useStyles from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WADialog/WADialog.styles'
import { WAOfferToCreateWallet } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WAOfferToCreateWallet/WAOfferToCreateWallet'
import { useToggleValue } from 'hooks/useToggleValue'
import { WithdrawalAddressForm } from './WAForm'

export const WithdrawalAddressCreate = () => {
  const [isCreateWalletDialogVisible, toggleIsCreateWalletDialogVisible] =
    useToggleValue()
  const classes = useStyles({ isCreateWalletDialogVisible })
  const hint = (
    <WAOfferToCreateWallet onClick={toggleIsCreateWalletDialogVisible} />
  )

  return (
    <>
      <WADialog
        open
        classes={{
          paper: classes.dialog,
          root: classes.dialog
        }}
      >
        <WADialogTitle label='Add Blockchain Address' />
        <WADialogContent>
          <WithdrawalAddressForm hint={hint} />
        </WADialogContent>
      </WADialog>
      <CreateWalletDialog
        classes={{
          paper: classes.createDialog,
          root: classes.createDialog
        }}
        open
        onClose={toggleIsCreateWalletDialogVisible}
      >
        <CreateWalletDialogTitle
          label='Create Wallet'
          onButtonCloseClick={toggleIsCreateWalletDialogVisible}
        />
        <CreateWalletDialogContent />
      </CreateWalletDialog>
    </>
  )
}
