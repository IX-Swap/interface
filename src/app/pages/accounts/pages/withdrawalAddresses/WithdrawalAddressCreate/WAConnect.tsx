import { FileCopyOutlined } from '@mui/icons-material'
import {
  Alert,
  CircularProgress,
  DialogActions,
  Grid,
  IconButton,
  InputAdornment
} from '@mui/material'
import { LoadingIndicator } from 'app/components/LoadingIndicator/LoadingIndicator'
import { WalletConnectionStatus } from 'app/pages/accounts/pages/withdrawalAddresses/hooks/useConnectMetamaskWallet'
import { WAConnectActions } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WAConnectActions'
import { WAInfoFields } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WAInfoFields'
import { TypedField } from 'components/form/TypedField'
import { privateClassNames } from 'helpers/classnames'
import { copyToClipboard } from 'helpers/clipboard'
import React, { ReactElement, useEffect } from 'react'
import { useFormContext } from 'react-hook-form'
import { WithdrawalAddressFormValues } from 'types/withdrawalAddress'
import { TextInput } from 'ui/TextInput/TextInput'
import { useDebouncedCallback } from 'use-debounce'
import { useCheckAddress } from '../hooks/useCheckAddress'
import { WAPair } from './WAPair'

export interface WAConnectProps {
  hint: ReactElement
  status: WalletConnectionStatus
  getAccount: (walletAddress: string) => Promise<void>
}

export const WAConnect = ({ hint, status, getAccount }: WAConnectProps) => {
  const { watch, control } = useFormContext<WithdrawalAddressFormValues>()
  const address = watch('address')
  const wallet = watch('wallet')
  const [checkAddress, { isLoading, data, isSuccess }] = useCheckAddress({
    onSuccess: values => {
      control.setValue('wallet', values.data?.walletApplication)
      control.setValue('network', values.data?.networkId)
    }
  })
  const { callback: debouncedCheckAddress } = useDebouncedCallback(
    checkAddress,
    300
  )

  useEffect(() => {
    if (address?.trim().length > 0) {
      void debouncedCheckAddress(address)
    }
  }, [address, debouncedCheckAddress])

  const isInitialising = status === WalletConnectionStatus.INITIALISING
  const hasWallet =
    status === WalletConnectionStatus.INITIALISED ||
    status === WalletConnectionStatus.VERIFYING ||
    status === WalletConnectionStatus.ERROR
  const isVerifying = status === WalletConnectionStatus.VERIFYING
  const isVerified = status === WalletConnectionStatus.SUCCESS
  const allowConnect = data?.data.allowConnect ?? false

  return (
    <>
      <Grid item mt={3}>
        <TypedField
          className={privateClassNames()}
          component={TextInput}
          control={control}
          name='address'
          label='Blockchain Address'
          variant='outlined'
          disabled={allowConnect}
          InputProps={{
            endAdornment: isLoading ? (
              <CircularProgress size={18} thickness={6} />
            ) : (
              allowConnect && (
                <InputAdornment position='end'>
                  <IconButton
                    size='small'
                    onClick={() => copyToClipboard(address)}
                  >
                    <FileCopyOutlined color='action' />
                  </IconButton>
                </InputAdornment>
              )
            )
          }}
        />
      </Grid>
      <Grid item xs={12}>
        <WAPair wallet={wallet} networkCode={data?.data.code} />
      </Grid>
      {isSuccess && !allowConnect && (
        <Grid item xs={12}>
          <Alert severity='error'>
            We cannot connect your wallet at the moment. Connection of other
            wallets to the platform is coming soon.
          </Alert>
        </Grid>
      )}
      {hasWallet && <WAInfoFields />}
      {!hasWallet && hint}
      <DialogActions>
        <Grid item container justifyContent='flex-end'>
          <WAConnectActions
            isVerifying={isVerifying}
            isVerified={isVerified}
            isLoading={isInitialising || isVerifying}
            hasWallet={hasWallet}
            allowConnect={allowConnect}
            getAccount={async () => await getAccount(address)}
          />
        </Grid>
      </DialogActions>

      {isInitialising && (
        <LoadingIndicator
          opacity={1}
          title='Initializing. Please Wait...'
          message='By connecting a wallet, you agree to InvestaXâ€™ Terms and Conditions and acknowledge that you have read and understand the InvestaX Privacy Policy.'
        />
      )}

      <input {...control.register('wallet')} hidden />
      <input {...control.register('network')} hidden />
    </>
  )
}
