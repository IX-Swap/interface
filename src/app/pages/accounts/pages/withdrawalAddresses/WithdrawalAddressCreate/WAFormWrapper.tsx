import React from 'react'
import { Grid } from '@material-ui/core'
import { Submit } from 'components/form/Submit'
import { WithdrawalAddressForm } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WAForm'
import { WAFormFields } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WAFormFields'
import { WADialogActions } from 'app/pages/accounts/pages/withdrawalAddresses/WADialog/WADialogActions'

export const WAFormWrapper = () => {
  return (
    <WithdrawalAddressForm defaultValues={{ agree: false }}>
      <Grid container direction='column' spacing={3}>
        <WAFormFields />
        <WADialogActions>
          <Grid item container justify='center'>
            <Submit color='primary' variant='contained'>
              Submit
            </Submit>
          </Grid>
        </WADialogActions>
      </Grid>
    </WithdrawalAddressForm>
  )
}
