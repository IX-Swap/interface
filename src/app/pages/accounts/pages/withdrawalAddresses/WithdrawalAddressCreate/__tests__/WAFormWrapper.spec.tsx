import userEvent from '@testing-library/user-event'
import MockAdapter from 'axios-mock-adapter'
import { accountsURL, blockchainNetworksURL } from 'config/apiURL'
import { act } from 'react-dom/test-utils'
import web3Service from 'services/web3'
import { render, snackbarServiceMock, waitFor, within } from 'test-utils'
import { networks } from '__fixtures__/network'
import { WAFormWrapper } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WAFormWrapper'
import { user } from '__fixtures__/user'
import * as useAuth from 'hooks/auth/useAuth'
import { BlockchainWallet } from 'components/form/WalletSelect'

/* eslint-disable import/first */
jest.deepUnmock('axios')
import { _axios } from 'services/api'

jest.spyOn(useAuth, 'useAuth').mockReturnValue({ user, isAuthenticated: true })

const getAccountSpy = jest.spyOn(web3Service, 'getAccount')
const signWalletSpy = jest.spyOn(web3Service, 'signWallet')

const mock = new MockAdapter(_axios)
const network = networks[0]
const walletLabel = 'My Metamask Wallet'
const walletAddress = '5cc2c1a506a7b202ecb157de5383c084'
const walletHash = '02b7bd615871b79ec9d9d5e75bc70f56'
const signedHash = '4de01fc6170f8eaa9af35752a31dd989'

describe('WAFormWrapper', () => {
  beforeEach(() => {
    mock.onGet(accountsURL.withdrawalAddresses.getAllNetworks).reply(200, {
      data: networks,
      message: 'OK'
    })

    mock
      .onPost(blockchainNetworksURL.generateWalletHash, {
        walletAddress
      })
      .reply(200, {
        data: walletHash
      })

    mock
      .onPost(blockchainNetworksURL.verifyWalletOwnership, {
        walletAddress,
        signedHash
      })
      .reply(200, {
        data: {
          isVerified: true
        }
      })

    mock
      .onPost(accountsURL.withdrawalAddresses.create(user._id), {
        address: walletAddress,
        label: walletLabel,
        wallet: BlockchainWallet.Metamask,
        network: network._id
      })
      .reply(200, {})
  })

  it('should work', async () => {
    const { getByTestId, getByRole, getByText, getByLabelText, container } =
      render(<WAFormWrapper />, { mockAPI: false })

    // select connect option
    userEvent.click(getByText('Connect to Wallet'))

    // open networks dropdown
    const networkSelect = getByTestId('network-select')
    userEvent.click(within(networkSelect).getByRole('button'))

    // select first network
    const networksDropdown = getByRole('listbox')
    const ropstenNetwork = await waitFor(() =>
      within(networksDropdown).getByText(network.name)
    )
    userEvent.click(ropstenNetwork)

    // select wallet
    const walletSelect = getByTestId('wallet-select')
    userEvent.click(within(walletSelect).getByRole('button'))

    // select first network
    const walletsDropdown = getByRole('listbox')
    userEvent.click(within(walletsDropdown).getByText('Metamask'))

    // click connect wallet
    getAccountSpy.mockResolvedValueOnce(walletAddress)
    userEvent.click(getByText('Connect Wallet'))

    // check for form values
    await waitFor(() =>
      expect(getByTestId('blockchain-address-form')).toHaveFormValues({
        variant: 'connect',
        network: network._id,
        address: walletAddress
      })
    )

    // click sign challenge
    signWalletSpy.mockResolvedValueOnce(signedHash)
    await act(async () => {
      userEvent.click(getByText('Sign Challenge'))
    })

    // fill in address label
    userEvent.type(getByLabelText('Address Label'), walletLabel)

    // agree to the terms
    userEvent.click(getByRole('checkbox'))

    // check for form values
    await waitFor(() =>
      expect(getByTestId('blockchain-address-form')).toHaveFormValues({
        variant: 'connect',
        network: network._id,
        address: walletAddress,
        wallet: BlockchainWallet.Metamask,
        agree: true,
        label: walletLabel
      })
    )

    // record form snapshot
    expect(container).toMatchSnapshot()

    const submitButton = await waitFor(() => getByText('Submit'))
    userEvent.click(submitButton)

    await waitFor(() => {
      expect(snackbarServiceMock.showSnackbar).toBeCalledWith(
        'Success',
        'success'
      )
    })
  })
})
