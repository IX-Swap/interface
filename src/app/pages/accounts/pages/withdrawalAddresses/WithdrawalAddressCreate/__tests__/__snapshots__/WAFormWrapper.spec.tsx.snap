// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WAFormWrapper should work 1`] = `
<div>
  <form
    data-testid="blockchain-address-form"
    style="width: 100%;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiFormGroup-root"
          control="[object Object]"
          id="variant"
          label="Wallet Type"
          path="variant"
          role="radiogroup"
          style="flex-direction: row; justify-content: space-between;"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-4"
                  name="variant"
                  type="radio"
                  value="connect"
                />
                <div
                  class="PrivateRadioButtonIcon-root-5 PrivateRadioButtonIcon-checked-7"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Connect to Wallet
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-4"
                  name="variant"
                  type="radio"
                  value="create"
                />
                <div
                  class="PrivateRadioButtonIcon-root-5"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Input Blockchain Address
            </span>
          </label>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
            for="network"
          >
            Blockchain Network
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
            data-testid="network-select"
            style="min-width: 70px;"
          >
            <div
              aria-disabled="true"
              aria-haspopup="listbox"
              aria-labelledby="network"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled Mui-disabled"
              id="network"
              role="button"
            >
              Ropsten Test Network (Public Ethereum)
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              name="network"
              tabindex="-1"
              value="5f88035d7ae447ee9274d4fa"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined Mui-disabled"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-10 PrivateNotchedOutline-legendNotched-11"
              >
                <span>
                  Blockchain Network
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
            for="wallet"
          >
            Select Wallet
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
            data-testid="wallet-select"
          >
            <div
              aria-disabled="true"
              aria-haspopup="listbox"
              aria-labelledby="wallet"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled Mui-disabled"
              id="wallet"
              role="button"
            >
              <div
                class="MuiBox-root MuiBox-root-16"
              >
                <img
                  alt="Metamask"
                  src="metamask_another.png"
                />
                <span
                  class="MuiBox-root MuiBox-root-17"
                >
                  Metamask
                </span>
              </div>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              name="wallet"
              tabindex="-1"
              value="METAMASK"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined Mui-disabled"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-10 PrivateNotchedOutline-legendNotched-11"
              >
                <span>
                  Select Wallet
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-13 MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
            for="address"
          >
            Wallet Address
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root fs-exclude"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-disabled Mui-disabled"
              data-shrink="false"
              for="address"
              id="address-label"
            >
              Wallet Address
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                disabled=""
                id="address"
                name="address"
                type="text"
                value="5cc2c1a506a7b202ecb157de5383c084"
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    Wallet Address
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
            for="label"
          >
            Address Label
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root fs-exclude"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="label"
              id="label-label"
            >
              Address Label
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="label"
                name="label"
                type="text"
                value="My Metamask Wallet"
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10 PrivateNotchedOutline-legendNotched-11"
                >
                  <span>
                    Address Label
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="memo"
          >
            Memo
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root fs-exclude"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="memo"
              id="memo-label"
            >
              Memo
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="memo"
                name="memo"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    Memo
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <label
          class="MuiFormControlLabel-root"
          id="agree"
          path="agree"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root makeStyles-root-22 PrivateSwitchBase-checked-2 Mui-checked"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                aria-label="decorative checkbox"
                class="PrivateSwitchBase-input-4"
                data-indeterminate="false"
                name="agree"
                type="checkbox"
                value=""
              />
              <span
                class="makeStyles-icon-23 makeStyles-checkedIcon-24"
              />
            </span>
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <p
              class="MuiTypography-root makeStyles-text-21 MuiTypography-body1 MuiTypography-colorInherit"
            >
              I understand and agree that InvestaX will check this address against fradulent activities.
            </p>
          </span>
        </label>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-flex-end"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
